rewrite-core/src/main/java/org/openrewrite/Result.java

```diff
@@ -255,4 +260,13 @@ public String diff(@Nullable Path relativeTo, PrintOutputCapture.@Nullable Marke
     public String toString() {
         return diff();
     }
+
+    public static boolean isLocalAndHasNoChanges(@Nullable SourceFile before, @Nullable SourceFile after) {
+        return (before == after) ||
+                (before != null && after != null &&
+                        // Remote source files are fetched on `printAll`, let's avoid that cost.
+                        !(before instanceof Remote) &&
+                        !(after instanceof Remote) &&
```

Comment:
Would put this in one line:
```suggestion
                        !(before instanceof Remote) && !(after instanceof Remote) &&
```

If you read it, it's easier to follow:
```
BEFORE and AFTER are the same
<OR>
BEFORE and AFTER are not null <AND>
  BEFORE and AFTER are no instance of Remote <AND>
  BEFORE.printAll() is the same as AFTER.printAll()
```